CMSgov/cms-carts-seds
# Move deploy to run again (fixed)

### Description

Fix CARTS running via run wrapper

We had to revert the original change because the kafka deploy was hanging and timing out. Here I played around with the configuration to solve this issue.
I added awaits to more of the steps, but it was still hanging.
Ultimately what fixed it was changing the carts-bigmac-streams hook script in the serverless file to use the `aws lambda invoke` cli instead of `sls invoke`

When we migrated to serverless compose before we had issues with these hooks. Although we deploy the kafka service independently, I'm guessing something about our runner has an issue when you try to invoke a sls command within a sls command. By using the aws invoke, it's able to complete!

[Original PR](https://github.com/Enterprise-CMCS/macpro-mdct-carts/pull/139679)
[Revert PR](https://github.com/Enterprise-CMCS/macpro-mdct-carts/pull/139687)

### Related ticket(s)


CMDCT-3820

---
### How to test

Notice that the kafka deploy was hanging for a long time (manually canceled) in [this run](https://github.com/Enterprise-CMCS/macpro-mdct-carts/actions/runs/9979101601/job/27577496876)

Verify CARTS and the kafka service deployed successfully in [this run](https://github.com/Enterprise-CMCS/macpro-mdct-carts/actions/runs/9979654423/job/27579227796)

Note: the latest deploy will not have the kafka deploy because I had to change the `if` statement back to just main branches. The above linked runs represent the second and third commit deploys.

I verified the stacks deployed in AWS as well. Oddly enough they deployed when the runner was hanging as well, so it really did just get caught on that script hook.

### Notes


I modified the `if` logic to let the kafka deploy run in this branch until it worked. That logic is now reverted to only apply to the main branches.

---
### Pre-review checklist
